docker/hyperdx/Dockerfile

```diff
@@ -0,0 +1,143 @@
+COPY --from=app ./public ./public 
+COPY --from=app ./styles ./styles
+
+ENV NEXT_TELEMETRY_DISABLED 1
+ENV NEXT_OUTPUT_STANDALONE false
+ENV NEXT_PUBLIC_IS_LOCAL_MODE false
+RUN yarn build && rm -rf node_modules && yarn workspaces focus --production
+
+
+# prod ############################################################################################
+FROM node:${NODE_VERSION}-alpine AS prod
+
+ENV NODE_ENV production
+
+# Install libs used for the start script
+RUN npm install -g concurrently@9.1.0
+
+USER node
+
+# Set up API
+WORKDIR /app/api
+COPY --chown=node:node --from=api_builder ./app/api/build ./build
+COPY --chown=node:node --from=api_builder ./app/api/node_modules ./node_modules
+
+# Set up App
+WORKDIR /app/app
+COPY --from=app_builder /app/app/next.config.js ./
+COPY --chown=node:node --from=app_builder /app/app/public ./public
+COPY --chown=node:node --from=app_builder /app/app/.next ./.next
+COPY --from=app_builder /app/app/node_modules ./node_modules
+COPY --from=app_builder /app/app/package.json ./package.json
+
+# Set up start script
+COPY --chown=node:node --from=hyperdx ./entry.prod.sh /etc/local/entry.sh
+ENTRYPOINT ["sh", "/etc/local/entry.sh"]
+
+# all-in-one base ############################################################################################
+FROM scratch AS all-in-one-base
+
+# Copy from clickhouse and otel collector bases
+COPY --from=clickhouse_base / /
+COPY --from=otel_collector_base /otelcol-contrib /usr/local/bin/otelcol-contrib
+
+# Copy Node.js runtime from node base
+COPY --from=node_base --link /usr/local/bin /usr/local/bin
+COPY --from=node_base --link /usr/local/lib /usr/local/lib
+COPY --from=node_base /usr/lib /usr/lib
+COPY --from=node_base /usr/local/include /usr/local/include
+
+# Set up Clickhouse
+COPY --from=clickhouse ./local/*.xml /etc/clickhouse-server
+COPY --from=hyperdx ./clickhouseConfig.xml /etc/clickhouse-server/config.xml
+
+# Set up Otel Collector
+COPY --from=otel-collector ./config.yaml /etc/otelcol-contrib/config.yaml
+
+# Install MongoDB and other dependencies (consolidated into a single RUN command)
+RUN echo 'http://dl-cdn.alpinelinux.org/alpine/v3.9/main' >> /etc/apk/repositories && \
+  echo 'http://dl-cdn.alpinelinux.org/alpine/v3.9/community' >> /etc/apk/repositories && \
+  apk update && \
+  apk add --no-cache mongodb yaml-cpp=0.6.2-r2 curl && \
+  mkdir -p /data/db && \
+  rm -rf /var/cache/apk/*
+
+ENV NODE_ENV production
+
+# Set up App (copy from prod stage)
+COPY --from=prod /app/api /app/api
+COPY --from=prod /app/app /app/app
+COPY --from=hyperdx ./entry.local.sh /etc/local/entry.base.sh
+
+WORKDIR /app/app
+
+# Add hosts entry in entrypoint script instead of here
+
+# Expose ports
+EXPOSE 8080 4317 4318 13133 8123 9000
+
+# all-in-one no auth ############################################################################################
+FROM all-in-one-base AS all-in-one-noauth
+COPY --from=hyperdx ./entry.local.noauth.sh /etc/local/entry.sh
```

Comment:
The auth/noauth local files only set a single, environment variable. Could we just use the `ENV` command here to set the value and avoid additional shell script files?

Reply:
The whole point is to prevent users from overriding the environment variable to potentially disable auth